lib/puppet/ssl/host.rb

```diff
@@ -568,14 +568,14 @@ def check_for_certificate_on_disk(cert_name)
   def download_certificate_from_ca(cert_name)
     return nil if Puppet::SSL::Host.ca_location != :remote
 
-    response = Puppet::Rest::Routes.get_certificate(http_client, cert_name)
-    if response.ok?
-      body = response.read_body
-      begin
-        Puppet::SSL::Certificate.from_s(body)
-      rescue OpenSSL::X509::CertificateError
-        raise Puppet::Error, _("Response from the CA did not contain a valid certificate for %{cert_name}.") % { cert_name: cert_name }
-      end
+    begin
+      cert = Puppet::Rest::Routes.get_certificate(http_client, cert_name)
+      Puppet::SSL::Certificate.from_s(cert)
+    rescue Puppet::Rest::ResponseError
```

Comment:
Could this potentially be swallowing other responses that are not `No cert for %{cert_name}` kind of errors?

Reply:
Yeah, I suppose it could. The exception this wraps gets thrown for any non-200 response. Might be worth checking for a 404 response code and re-throwing otherwise?